src/hooks/screen/useChatScreenHook.ts

```diff
@@ -27,35 +30,60 @@ interface ScrollContextProps {
 export const ScrollContext = React.createContext<ScrollContextProps | null>(null);
 
 function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
-  const {localDb, chat, refresh} = useLocalDatabaseHook();
+  const {localDb, refresh, otherListener, refreshWithId, chat} = useLocalDatabaseHook();
   const {selectedChannel, goBackFromChatScreen, goToChatInfoScreen} = useChatUtilsHook();
   const {anonProfileId} = useUserAuthHook();
+  const {queue} = useDatabaseQueueHook();
+
+  const isInitiatingChatRef = React.useRef(false);
 
   const [selfAnonUserInfo, setSelfAnonUserInfo] = React.useState<any>(null);
   const [chats, setChats] = React.useState<ChatSchema[]>([]);
+
   const initChatData = async () => {
     if (!localDb || !selectedChannel) return;
+    const chatListener = otherListener[`chat_${selectedChannel?.id}`];
+    if (!chatListener) return;
+    // console.log('checkpoint chat screen 1');
+
     try {
+      isInitiatingChatRef.current = true;
       const myUserId = await getUserId();
       const myAnonymousId = await getAnonymousUserId();
-      const data = (await ChatSchema.getAll(
-        localDb,
-        selectedChannel?.id,
-        myUserId,
-        myAnonymousId
-      )) as ChatSchema[];
-      setChats(data);
 
-      if (type === 'ANONYMOUS') {
-        const userInfo = await UserSchema.getSelfAnonUserInfo(
-          localDb,
-          anonProfileId,
-          selectedChannel?.id
-        );
+      queue.addPriorityJob({
+        priority: QueueJobPriority.HIGH,
+        label: `get-all-chat-${selectedChannel?.name}`,
+        task: async () => {
+          const data = (await ChatSchema.getAll(
+            localDb,
+            selectedChannel?.id,
+            myUserId,
+            myAnonymousId
+          )) as ChatSchema[];
+          setChats(data);
+          isInitiatingChatRef.current = false;
+
+          return data;
+        }
+      });
 
-        setSelfAnonUserInfo(userInfo);
+      if (type === 'ANONYMOUS') {
+        queue.addPriorityJob({
+          priority: QueueJobPriority.MEDIUM,
+          label: `get-anon-user-info-${selectedChannel?.name}`,
+          task: async () => {
+            const userInfo = await UserSchema.getSelfAnonUserInfo(
+              localDb,
+              anonProfileId,
+              selectedChannel?.id
+            );
+            setSelfAnonUserInfo(userInfo);
+          }
+        });
       }
     } catch (e) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChatData` function has been updated to use a queue for asynchronous tasks. This is a good practice as it allows better control over the execution of asynchronous tasks and can help prevent potential race conditions. However, there's a potential issue with error handling. In the catch block, you're only logging the error and not doing anything else. It would be beneficial to handle these errors in a more user-friendly way, such as showing an error message to the user or retrying the operation.

```diff
     } catch (e) {
      isInitiatingChatRef.current = false;
+       SimpleToast.show('Error initializing chat data. Please try again.');
       console.log(e, 'error get all chat');
     }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->